### ISSUE TITLE
Various JLine inconsistencies

### ISSUE THREAD

**Issue by stewSquared** [lihaoyi/Ammonite#354]
GNU/readline is great.

JLine as it's used in the official scala repl, works well for the most part. I'll run into obscure inconsistencies. <alt+t>, for example, does nothing where I expect it to transpose two words. This isn't ideal, but it's close enough not to bother me.

Now, for whatever reason, JLine as used by ammonite has many more inconsistencies - enough to be rather frustrating. It's like going to heaven only to stub your toe every five minutes.

`Alt-t` moves the cursor forward two-spaces, but continues inputing text as if the cursor never moved. Unless you press this at the beginning of a line, then Ammonite crashes and drops you into a plain scala repl. The solution is to `Ctrl-c` my current expression. In scala, nothing happens. In readline, two words are transposed.

Various other key combos result in similar behaviour. I often find myself with a displaced cursor without knowing exactly why that is the case.

`Ctrl-t` abandons my current line and enters "Char Display Mode, whatever that is.
In scala, two characters are transposed, as expected.

`Alt-f` almost works as expected. It's supposed to jump to the end of the current word, unless you're already there, in which case you jump to the end of the next word. Ammonite will jump to the end of the next word if you're one character away from the end of the current word. This makes for confusing behavior with one-letter words.

`Alt-d` will delete the next word instead of the rest of the current word, if you're one away from the end of the current word. I've lost many parens this way.

That's just a few, but they are all things I noticed in my first few minutes of using Ammonite. The greater theme here is that ammonite doesn't behave like scala-repl, even though they both use JLine, and the newcomer would have a nicer experience if the repl didn't just crash after certain common key-combos.

**Comment by lihaoyi** [comment]
If you want "raw" Jline behavior, you can cut over to using it via

```scala
@ repl.frontEnd() = ammonite.repl.frontend.FrontEnd.JLineUnix
```

Otherwise you're using a custom thing which has a bunch of benefits (syntax highlighting, multiline-editing, ...) albeit at a cost of re-implementing parts of jline, sometimes imperfectly.

@stewSquared if you're interested in cleaning up this issue it would be good to have an issue with short, easy-to-repro cases where the behavior diverges. This issue is too wordy and has too much irrelevant prose.

It shouldn't be hard to fix them all, someone just has to notice them and report it. 

I could also close this issue and you can open a new, cleaner one if you'd prefer

**Comment by stewSquared** [comment]
@lihaoyi Go ahead and close this one. I'll start adding smaller, cleaner issue as I play with Ammonite and map out the precise differences.

**Comment by lihaoyi** [comment]
Ok cool!
